src/v2/components/database/builder.ts

```diff
@@ -0,0 +1,122 @@
+import * as pulumi from '@pulumi/pulumi';
+import * as aws from '@pulumi/aws';
+import * as awsx from '@pulumi/awsx';
+import { Database } from '.';
+
+export namespace DatabaseBuilder {
+  export type Config = Omit<
+    Database.Args,
+    | 'vpc'
+    | 'enableMonitoring'
+    | 'parameterGroupName'
+    | 'customParameterGroupArgs'
+    | 'kmsKeyId'
+    | 'snapshotIdentifier'
+  >;
+}
+
+export class DatabaseBuilder {
+  private _name: string;
+  private _config?: DatabaseBuilder.Config;
+  private _vpc?: Database.Args['vpc'];
+  private _enableMonitoring?: Database.Args['enableMonitoring'];
+  private _parameterGroupName?: Database.Args['parameterGroupName'];
+  private _customParameterGroupArgs?: Database.Args['customParameterGroupArgs'];
+  private _kmsKeyId?: Database.Args['kmsKeyId'];
+  private _snapshotIdentifier?: Database.Args['snapshotIdentifier'];
+
+  constructor(name: string) {
+    this._name = name;
+  }
+
+  public configure(
+    dbName: DatabaseBuilder.Config['dbName'],
+    username: DatabaseBuilder.Config['username'],
+    config: Omit<DatabaseBuilder.Config, 'dbName' | 'username'> = {},
+  ): this {
+    this._config = {
+      dbName,
+      username,
+      ...config,
+    };
+
+    return this;
+  }
+
+  public withVpc(vpc: pulumi.Input<awsx.ec2.Vpc>): this {
+    this._vpc = pulumi.output(vpc);
+
+    return this;
+  }
+
+  public withMonitoring(): this {
+    this._enableMonitoring = true;
+
+    return this;
+  }
+
+  public createFromSnapshot(snapshotIdentifier: pulumi.Input<string>): this {
+    this._snapshotIdentifier = snapshotIdentifier;
+
+    return this;
+  }
+
+  public useExistingParameterGroup(
+    parameterGroupName: pulumi.Input<string>,
+  ): this {
+    this._parameterGroupName = parameterGroupName;
+
+    return this;
+  }
+
+  public withCustomParameterGroup(
+    customParameterGroupArgs: pulumi.Input<aws.rds.ParameterGroupArgs>,
+  ): this {
+    this._customParameterGroupArgs = customParameterGroupArgs;
+
+    return this;
+  }
```

Comment:
Would it make sense to consolidate this with the method overload?
In that case we could have `withParameterGroup` that can receive args or name which should simplify developer experience as they don't have to think about which method to choose, or try to understand different methods.